### PR TITLE
Added copyright notice for Finnish translation

### PR DESCRIPTION
**Pull request by fergusq**
I kindly ask you to include this addition to the copyright notice of the database.

Since the Finnish translation is a separate work from the English dictionary, I have copyright to it. To protect my interests in the (currently unlikely) scenario that Paramount, CBS or other commercial organization wanted to use my Finnish dictionary, I want to license my work under a separate license that prevents commercial usage unless negotiated with me.

This license will not prevent you, boQwI', the Klingon Language Institute or other non-profit/non-commercial users from using the Finnish dictionary.

### CODE REVIEW

**Comment by De7vID** [comment]
The entire project (repository) is covered by one license, namely the Apache License, version 2.0. To apply different licenses to a subset of the data is too confusing. I also can't retroactively change the license on people who are already using the data, and I have no way of informing them of the change. (I'm not aware of anyone actually using the Finnish data outside of the **boQwI'** apps and your scripts, but there are users who are including it as a side effect of using the database in their projects at all.)

I'm not opposed to having the Finnish (or any other language) translation under a different license, but before that can happen, the database would need to be compartmentalised in such a way that a user would be able to pull into their projects only the data covered by licensing terms that they agree to (or, equivalently, be able to easily exclude the Finnish data), but that currently isn't possible. (For example, it would work if each language was under its own directory, say `data_fi` for Finnish, and there's a separate LICENSE file that applies just to that directory.)

The way your change is written, a user can easily misread that the entire database is under the Creative Commons license, since it would be the last line of the copyright notice. I don't think the header is the right place for the notice, but I can't think of a better place, unless we add a section to the README.md (in which case we'd need to mention the licensing for the rest of the database). Another possibility is to put a notice with the Finnish credits (i.e., after your name) in the "about" page. But maybe it would be too much clutter there. (What if Lieven and Zrajm want to do the same things for German and Swedish, respectively, which is mostly their work?) Wherever the copyright notice goes, though, it needs to be very clear what it applies to (and what it doesn't apply to).

**Comment by fergusq** [comment]
I will make the necessary changes to compartmentalise the database.

**Comment by De7vID** [comment]
How do you plan to do that?

**Comment by fergusq** [comment]
Each entry must have a unique id that does not change when the homonym numbers or entry order is changed. I believe this will help Zrajm too with the Klingonska dictionary.

In the first phase, I will add such an id to each entry.

In the second phase, I will move `<definition_xx>`, `<notes_xx>` and `<search_tags_xx>` fields to separate files for each language except English.

Then, I'll create a new script (merge_langs.py) that will read all the different files and concatenate them to a single XML file. generate_db.sh and xml2json.py must call this script.

For convenience, there should be scripts that sync the different languages so that when entries are added, machine translated entries are added to the translated files.

**Comment by thiagocmatos** [comment]
If you split all languages to its own fies, you will need to o update all scripts and we will also lose the ability to fast translate existing entries. It will be way much harder to submit new translations for other languages.

**Comment by fergusq** [comment]
Yes and no. This change will require us to adapt to a new workflow, which in itself isn't better or worse than the old one.

I will make a synchronization script. It will add all new untranslated entries to the new XML files. So when De'vID adds a new word, Finnish, Portuguese et al. entries will be automatically created. In some ways, this might even make editing the non-English words easier, as the XML files will have less noise from all the other languages (more entries fit to the screen).

The downside is that someone must remember to run the script after each batch of new words. I will make it so that the merge script will refuse to create the uninified XML unless the synchronization has been called.

Not all scripts must be updated: the ones that work on JSON work as they are. The ones that use XML must call the merge script instead of just concatenating the XML files.

**Comment by De7vID** [comment]
Better work with @zrajm as this will undoubtedly affect PR #654.

**Comment by De7vID** [comment]
Adding a unique ID to each entry is issue #199. Feel free to assign yourself if you're going to do it.

**Comment by fergusq** [comment]
I separated the translated entries to their own files and made necessary changes to xml2json.py and generate_db.sh. Other scripts need to be upgraded as well, but I'd like the changes to be reviewed before it.

@zrajm I have added ids to each entry. Are this suitable for your purposes?

**Comment by De7vID** [comment]
I think the subdirectories should have a prefix and not use the bare locale code. There are other subdirectories, so a prefix would group them together. Maybe `translations_xx[_yy]`, reflecting the way the fields are named in their contents?

How does a user exclude a translation, or what happens if a subdir is missing? Would it just result in a blank/empty `definition_fi` field or whatever? For example, it would be convenient if languages which are not under the same license are _not_ included by default, and a user has to explicitly include it as a git submodule. That way, it would be clear that the user has explicitly taken an action to include that language. As well, for people who are already using the existing data under the Apache license, the next time they sync the repo, the differently-licensed data will be removed and if they want to include it again, they'd have to take action to do so. So then it's very clear that they agree to the different license for that locale.

**Comment by fergusq** [comment]
I can make it so that the merge script takes the included languages as an argument so that the user can specify which languages are included in the merged xml (same for JSON).

I don't think the translations should be git submodules. It is very common for repositories to contain stuff under multiple licenses, and it's also very common that sometimes the licenses change. If the scripts change, they already have to take action by explicitly running the scripts differently. I'm not aware that there even are any people using the Finnish translations in this repository, so separating translations to different submodules would just be making things hard with no benefit.

**Comment by De7vID** [comment]
Okay, I think it'd be fine not to have submodules. The fact that the scopes of the licenses apply up to directory boundaries is clear enough (unlike the previous situation where they applied to different parts of one file, which was extremely confusing).

If the translation subdirectories are renamed as I suggested, and the existing scripts are updated to take this into account so existing users wouldn't have to make any changes, then I'm okay with this. It actually makes the K-E files a lot cleaner (it actually basically restores them to the way they used to be before I added German, then other languages). But please make sure that the existing scripts all work as intended.

But since I'm not the one adding most of the translations, what do the people contributing translations think of this change? @zrajm @LievenLitaer @thiagocmatos @Sigrlin @dadap opinions?

**Comment by fergusq** [comment]
When I'm ready I will close this pull request and open a new one, since it's easier than merging the changes that have been made to master to this branch.

How long should we wait for the other contributors to review these changes? I'd like to close this as soon as possible, but would be nice to hear the opinion of others.

**Comment by Sigrlin** [comment]
I wasn't deeply involved in anything except the word translation, so I don't have any comments.16:30, 29 августа 2022 г., De7vID ***@***.***>:
I don't know if they have no comments, have GitHub notifications turned off and so are unaware of this change, or are away from computer because they're on a deep space mission for the Klingon Empire.
Incidentally, I'm traveling for a month or so starting next week, so I'm very busy and may not have time or access to the hardware necessary to review this PR during that time. So there's no rush for this PR.
(Alternatively, if you really want it done before I leave, then it needs to be done ASAP, though I can't guarantee I have time for that.)

**Comment by dadap** [comment]
I explicitly renounce all copyright on any contributions I have made and will make to the klingon-assistant-data project. There is no need to get approval from me if the license of the project or of components of the project which I have contributed needs to change in the future.

My personal preference, if the license of a component which I have contributed to were to change, is that such a change would be to make it more permissive (e.g. BSD, MIT, release into the Public Domain), and not in the direction of a more restrictive license (e.g. GPL, Creative Commons with NC, ND, SA, or any combination thereof). However, as I have explicitly renounced any and all copyright claims, this preference need not be taken into consideration for any future licensing decisions.

FWIW, since restrictions on commercial use are incompatible with free software licenses, this would mean that projects which exclusively use free software would be prevented from using the Finnish translations of the boQwI' database. I am not aware of any free-software-only projects which currently use this database, much less any which require the Finnish translations, so this is a non-issue for practical purposes.

**Comment by thiagocmatos** [comment]
@fergusq, I can accept your file structure change but I I would suggest that the other language files also contains all English fields. it will make it way easier to provide translations and keep find mistakes.
Regarding copyright for the portuguese translations, I follow @dadap and "I explicitly renounce all copyright on any contributions I have made and will make to the klingon-assistant-data project".
@fergusq I need your advice on how to handle the Anki Decks in Finish. I am not sure if I can provide them anymore.

**Comment by De7vID** [comment]
Lieven also had some comments but he's on vacation and apparently can't reply here. He wants to know what the mechanism is to keep all the subdirectories (and main directory) in sync, and what the new process is to add a new German translation (i.e., a Germany translation for a new entry).

**Comment by De7vID** [comment]
> FWIW, since restrictions on commercial use are incompatible with free software licenses, this would mean that projects which exclusively use free software would be prevented from using the Finnish translations of the boQwI' database. I am not aware of any free-software-only projects which currently use this database, much less any which require the Finnish translations, so this is a non-issue for practical purposes.

This is a good point. **boQwI'** is [available on F-Droid](https://f-droid.org/en/packages/org.tlhInganHol.android.klingonassistant/) (see also [this issue](https://github.com/De7vID/klingon-assistant/issues/448)), which means that it has to comply with this [inclusion policy](https://f-droid.org/en/docs/Inclusion_Policy/). The app is built automatically at every tagged release. 

Now, since the _app itself_ is under the Apache license, this may be fine even if it is built from _sources_ which are under another license, but this is something we have to be careful about and check carefully.

**Comment by fergusq** [comment]
> I need your advice on how to handle the Anki Decks in Finish. I am not sure if I can provide them anymore.

There shouldn't be a problem since Anki is a non-commercial application. Or is there another reason (like Anki not allowing CC NC decks)?

> This is a good point. boQwI' is [available on F-Droid](https://f-droid.org/en/packages/org.tlhInganHol.android.klingonassistant/) (see also https://github.com/De7vID/klingon-assistant/issues/448), which means that it has to comply with this [inclusion policy](https://f-droid.org/en/docs/Inclusion_Policy/). The app is built automatically at every tagged release.

If I have understood correctly, this shouldn't be an issue. The app uses non-free data, but is itself free. You must tag it with "Anti-Feature: Non-free Assets".

Now, this is a kind of side-note, but none of the assets are free. The Apache License is null and void. The English database is an illegal pirated copy of TKD and KGT, and claiming it to have Apache License is just wishful thinking. The only reason it hasn't been a problem is that Paramount hasn't been interested in bullying us. It's not nice but it's how things are.

This brings us to the question about including the English definitions in the subdirectories as Thiago requested. I'm against it, since I want to distance my dictionary from the English one. In the unlikely case that Paramount were to send a cease and desist letter, I could say, the Finnish files have nothing to do with the English files at all. If they included the English definitions, they would contain copyrighted material as well, which would make them illegal as well. I understand that separating the files causes a lot of trouble to people, but legally speaking, I think it's the safest option for us.

**Comment by De7vID** [comment]
> The English database is an illegal pirated copy of TKD and KGT

This statement is blatantly untrue: the English database is not a _copy_ of TKD and KGT at all. It contains a lot more entries than are found in TKD and KGT, and leaves out a lot of their contents. 

If your claim is that it _violates_ the copyright of these works, I would say that's also untrue, but IANAL. Facts can't be copyrighted under American copyright law, only their expression. Since TKD and KGT are the only sources of facts about the definitions of many Klingon words, it is inevitable that anyone defining these words (i.e., stating the fact of their definition) will end up with definitions in many cases identical or very similar to what's in TKD or KGT (unlike a natural language where there can be more variation in how a word is defined). The question here is the degree to which the database more resembles a compilation of facts vs. a copy of an original expression of facts, and whether in the case of a conlang, the facts of definitions are not identical to their expression (see the "[merger doctrine](https://en.wikipedia.org/wiki/Idea%E2%80%93expression_distinction)" here). I believe the database to be a compilation of facts and thus not in violation of copyright, but whether my belief is correct will have to be (and can only be) determined by a court of law.

There's no legal precedent for copyrighting a language, and the amicus brief filed by the Language Creation Society stating the legal opinion that a language, even a conlang, cannot be copyrighted is widely held even if it hasn't been legally tested. This isn't settled law, so I'd appreciate it if you would refrain from stating an accusation of illegality as though it were a fact.

Of course, if some corporation were to claim otherwise in court, I would have neither the resources nor the time to defend against that claim whether it has merit or not. However, **boQwI'** is a KLI project, and there's a good relationship between the KLI and CBS/Paramount, so it would not be in the interest of the latter to test the legal waters in a case which might not even be decided in their favour.

btw, I am doubtful that the separation of the Finnish files from the English definitions will have any effect on their legal status pertaining to copyright (but, again, IANAL). There's clearly a 1:1 correspondence, and the selection of words in a dictionary falls under the "expression" side of the "idea-expression" distinction. So, either the fact that the database contains a different set of words from TKD and KGT makes it an original expression (and thus neither the English nor the Finnish lists fall under copyright for that reason), or the Finnish part of the database is an unauthorised translation of a copyrighted work, whether it's paired with the English definitions or not. Saying that the Finnish files "have nothing to do" with the English files at all is like saying that a Finnish translation of a book has nothing to do with the English original. 

Also, if you don't want the English definitions to be mirrored in the Finnish directory, that's fine. But since the German, Portuguese, and other language translations are under the same license and are subject to identical legal restrictions as the English, I don't see that as a blocker for mirroring the English definitions in the languages other than Finnish.

**Comment by fergusq** [comment]
I'm not a lawyer, but I work for a company that produces and licenses dictionaries, so I have some experience on the topic. While some parts of the English dictionary were written by you (or not part of TKD or KGT to begin with), and obviously those parts aren't copyright violations, a lot of the dictionary is a copy and this poses a legal threat. My intention is not to blame your or anyone here, this situation sucks and I'm as angry about it as everyone is. I only want that everyone here is aware of these issues.

> This statement is blatantly untrue: the English database is not a copy of TKD and KGT at all. It contains a lot more entries than are found in TKD and KGT, and leaves out a lot of their contents.

This is irrelevant. Copying only part and embedding it inside a larger whole is still copying.

> If your claim is that it violates the copyright of these works, I would say that's also untrue, but IANAL. Facts can't be copyrighted under American copyright law, only their expression. Since TKD and KGT are the only sources of facts about the definitions of many Klingon words, it is inevitable that anyone defining these words (i.e., stating the fact of their definition) will end up with definitions in many cases identical or very similar to what's in TKD or KGT (unlike a natural language where there can be more variation in how a word is defined). The question here is the degree to which the database more resembles a compilation of facts vs. a copy of an original expression of facts, and whether in the case of a conlang, the facts of definitions are not identical to their expression (see the "[merger doctrine](https://en.wikipedia.org/wiki/Idea%E2%80%93expression_distinction)" here). I believe the database to be a compilation of facts and thus not in violation of copyright, but whether my belief is correct will have to be (and can only be) determined by a court of law

It is established that dictionaries can be copyrighted (although single entries might not be copyrightable). This has two parts: 1) which words are included and 2) which words are used as translations. In this case, point 1 isn't met as the number of words is different. However, point 2 is certainly met: a large part of the entries are word-by-word copies and thus in violation of the copyright. Again, this is not my opinion, copyrightability of dictionaries is an established fact.

> There's no legal precedent for copyrighting a language, and the amicus brief filed by the Language Creation Society stating the legal opinion that a language, even a conlang, cannot be copyrighted is widely held even if it hasn't been legally tested. This isn't settled law, so I'd appreciate it if you would refrain from stating an accusation of illegality as though it were a fact.

This is not about copyrightability of languages. It's about copyrightability of dictionaries, which is a separate issue. The choice of which words to use as definitions/translations is copyrightable, and as Klingon words don't align in meaning with the English words, there is creative interpretation, which is a basis for copyright protection.

> Of course, if some corporation were to claim otherwise in court, I would have neither the resources nor the time to defend against that claim whether it has merit or not. However, boQwI' is a KLI project, and there's a good relationship between the KLI and CBS/Paramount, so it would not be in the interest of the latter to test the legal waters in a case which might not even be decided in their favour.

I hope this will never be tested in court, since, as you said, we don't have resources to fight in the American court system. However, I think it's important to understand the threats involved in this project.

> btw, I am doubtful that the separation of the Finnish files from the English definitions will have any effect on their legal status pertaining to copyright (but, again, IANAL). There's clearly a 1:1 correspondence, and the selection of words in a dictionary falls under the "expression" side of the "idea-expression" distinction. So, either the fact that the database contains a different set of words from TKD and KGT makes it an original expression (and thus neither the English nor the Finnish lists fall under copyright for that reason), or the Finnish part of the database is an unauthorised translation of a copyrighted work, whether it's paired with the English definitions or not. Saying that the Finnish files "have nothing to do" with the English files at all is like saying that a Finnish translation of a book has nothing to do with the English original.

The Finnish dictionary doesn't have a "1:1 correspondence". The choice of words is completely unique: each entry is written from the Finnish point of view and often don't use the "obvious" translations, if we can even talk about such things. boQwI' is only one source used in the Finnish dictionary, and I have used also canonical example usage and other sources to write the entries. One might argue that translations of whole sentences might be a violation, and that is a risk I'm willing to take. Normal single-word entries, however, are not in any way a violation of copyright.

> Also, if you don't want the English definitions to be mirrored in the Finnish directory, that's fine. But since the German, Portuguese, and other language translations are under the same license and are subject to identical legal restrictions as the English, I don't see that as a blocker for mirroring the English definitions in the languages other than Finnish.

I think this is a decision to be made by Thiago, Lieven, Zrajm, etc. after considering these threats presented.

**Comment by thiagocmatos** [comment]
I have a few questions.

How would be the flow to add a new translation on other language? can you give me an example? (I guess we will need to manualy copy the English Entry to the language file to the proper place, then make the translation. this is an unecessary manual step which may bring inconsistency problems.)

Keeping at least the english entries on the Portuguese files will be very very helpfull to create translations.

On the current file structure we have the huge advantage of finding very easily all changes made and since everything is on the same place it is very easy to keep consistancy. I am afraid that we will lose changes overview easily and miss changes. (not only new entries but also updates which are very frequenty (e.g. new info))

Regarding the Anki Decks, I will need to add the proper copyright to the Finish translation. someone may just copy the deck and use it and then it will infring your copyright. I need your advise on how to handle that. (maybe you prefer to support the Finish deck yourself, which would be ok for me).

About the copyright issue you mention, I am not an expert but I dont believe we will ever have a problem and even if someday some crazy thing happen, the problem will affect the whole project and all type of word lists so spliting the translations in files has not effect.

**Comment by De7vID** [comment]
> I'm not a lawyer, but I work for a company that produces and licenses dictionaries, so I have some experience on the topic.

There's a well-known cognitive bias in which people who are knowledgeable about a subject overestimate their understanding of related subjects. I accept that you may be knowledgeable about Finnish copyright laws pertaining to dictionaries. I think that you are overly confident in your apparent belief that copyright laws in other countries work the way they do in Finland, or that electronic databases of word definitions necessarily are treated under the law in the same way as dictionaries.

> This is irrelevant. Copying only part and embedding it inside a larger whole is still copying.

It _is_ relevant. You claimed that it is "an _illegal pirated_ copy". While copying a part of something is still copying, some types of copying are _legally permissible_ in copyright law. I'm not objecting to the "copy" part of your claim, but to the "illegal" part of your claim. I do not appreciate your making unfounded accusations of illegality.

> Again, this is not my opinion, copyrightability of dictionaries is an established fact.

No one is denying that dictionaries are copyrightable, _but how that copyright applies varies between countries_. In the US, where TKD and KGT were published, copyright law follows the doctrine of the idea-expression dichotomy. This says that ideas are not copyrightable, while expressions of those ideas can be. But if an idea can be expressed only in limited ways, something called the merger doctrine says that an expression cannot be protected. A dictionary definition is an expression, and may be subject to copyright. But a dictionary definition for a conlang might fall under the merger doctrine, because unlike a natural language there is only limited information available about each word, and thus a very small number of ways to express that information. This has not been tested in court, and is thus unsettled law. _That_ is what is relevant here.

Please stop making claims which are only true in Finland as though they apply to the rest of the world. Sometimes they do not. Sometimes they do, but they have different consequences.

Some jurisdictions follow what is called the sweat of the brow doctrine in copyright. The US is not one of them. If it were, the compilation of a Klingon word list, even if it included all the entries from TKD and KGT, would result in a work with a separate copyright. The EU _does_ follow this doctrine when it comes to databases, granting something called database rights which is different from copyright. The situation is quite complex, and isn't analogous to that of a dictionary whose entire origin is in Finland.

> The Finnish dictionary doesn't have a "1:1 correspondence". The choice of words is completely unique: each entry is written from the Finnish point of view and often don't use the "obvious" translations, if we can even talk about such things.

When I wrote "1:1 correspondence", I was referring to the fact that there is one entry in Finnish for each entry in Klingon (or English), and not to the closeness of the English and Finnish translations. If, as you wrote, "The choice of which words to use as definitions/translations is copyrightable", and as the Finnish database makes exactly the same choice of corresponding Klingon entries as the English database, then it would violate whatever copyright the English database violates (assuming that it does so) by virtue of being a translation. The fact that you used other sources in coming up with the translations, or wrote the definitions from an especially Finnish perspective, doesn't make it any less a translation. If someone translated Lord of the Rings into Finnish, but consulted Tolkien's other works in doing so, and made unique word choices because they're retelling the story from a culturally Finnish point of view, no lawyer or judge would accept the argument that it isn't still a translation.

> In the unlikely case that Paramount were to send a cease and desist letter, I could say, the Finnish files have nothing to do with the English files at all. If they included the English definitions, they would contain copyrighted material as well, which would make them illegal as well.

There's no possibility that your claim that the Finnish files have nothing to do with the English files at all would be upheld in a court of law. There's _clearly_ a 1:1 correspondence, in that each Finnish entry has an id (if we keep the discussed implementation) which uniquely identifies the corresponding Klingon-English entry. It is a translation (and it doesn't matter how original or unique or Finnish your choices are in choosing the words). If the non-Finnish portion of the **boQwI'** database does not violate any copyright by CBS/Paramount, and I (or the KLI) hold the copyright, then the Finnish translation is an authorised translation of the Klingon-English definitions. But if the non-Finnish portion of the **boQwI'** database _does_ violate some copyright, then the Finnish data does as well, by virtue of being an unauthorised translation of copyrighted material. The fact that it's in a separate directory is entirely irrelevant to this.

I think you've misjudged the nature of the legal threat. The database is a compilation of facts, some of which are expressions identical to expressions in copyrighted works. However, in the copyright jurisdiction of those works, copying is permissible under some circumstances, and copyright does not apply in others. In particular, copyright does not apply in the US when an idea has a limited number of expressions. You say that this isn't about the copyright of languages but the copyright of dictionaries, but in the case of a conlang like Klingon, where the entire language is defined by a small set of sources, the copyright of the dictionary _is_ closely related to the copyright of the language _in a copyright jurisdiction that follows the doctrine of the idea-expression dichotomy_, which the US is (and which Finland is not). And it is a widely held legal opinion that a conlang cannot be copyrighted. If this is true, one consequence would be that while The Klingon Dictionary falls under copyright, a definition like **batlh** "honored, with honor" or even **batlh Daqawlu'taH** "You will be remembered with honor" do not.

It's unlikely that CBS/Paramount will want to test this legal theory, since a definitive ruling that conlangs cannot be copyrighted, and that therefore copyright does not apply to (some) expressions of definitions of words in a conlang, is contrary to their interest. In fact, they had an opportunity to do so in the Axanar case but refrained from pushing this point. But they also don't need to have the law settled in their favour, because the mere threat of lawsuit (even if it turns out to be without merit) is enough to enforce compliance. 

In other words, there's no practical reason for CBS/Paramount to take legal action against a database compilation of Klingon-English definitions many of which come from their copyrighted works, because it is a widely held legal opinion that it is not a copyright violation (an opinion I also share), even though the law is not actually settled. _If_ it _is_ a copyright violation, then the Finnish database clearly is as well. If your purpose in putting the Finnish data in a separate directory and under a different license is to somehow magic away the fact that the Klingon-English data and Finnish data have the same copyright violation status with respect to the materials copyrighted by CBS/Paramount, I don't think that's going to work.

**Comment by fergusq** [comment]
> It is relevant. You claimed that it is "an illegal pirated copy". While copying a part of something is still copying, some types of copying are legally permissible in copyright law. I'm not objecting to the "copy" part of your claim, but to the "illegal" part of your claim. I do not appreciate your making unfounded accusations of illegality.

I see how that choice of words may have been insulting to you. I apologize. It was not my intention to blame you of anything, since I sincerely believe that there is a copyright violation. I was not even aware that you might disagree with me, since the case is so clear based on my experience. I now understand how it was not clear and want to make sure that it's not my intention to accuse you of anything. I'm only interested in protecting this project since it's important to me that there are no legal consequences for me in participating in it.

> Please stop making claims which are only true in Finland as though they apply to the rest of the world. Sometimes they do not. Sometimes they do, but they have different consequences.

As I'm resided in Finland, I face consequences primarily in the Finnish courts. I hope you understand that I must consider the legality of my actions through the Finnish legal system.

The English part of the dictionary is a superset of a clear word-by-word copy of TKD and KGT entries and thus it clearly is a violation of copyright (in Finland). Now, maybe the merger doctrine applies here and it would be allowed in the US. I don't know about that and to my knowledge conlang dictionaries haven't been tested in court. Copyright violation is a threat and I hope you understand that even if you are sure about your legal interpretation, I'm not, so I must consider these possibilities.

> There's no possibility that your claim that the Finnish files have nothing to do with the English files at all would be upheld in a court of law. There's clearly a 1:1 correspondence, in that each Finnish entry has an id (if we keep the discussed implementation) which uniquely identifies the corresponding Klingon-English entry. It is a translation (and it doesn't matter how original or unique or Finnish your choices are in choosing the words). If the non-Finnish portion of the boQwI' database does not violate any copyright by CBS/Paramount, and I (or the KLI) hold the copyright, then the Finnish translation is an authorised translation of the Klingon-English definitions. But if the non-Finnish portion of the boQwI' database does violate some copyright, then the Finnish data does as well, by virtue of being an unauthorised translation of copyrighted material. The fact that it's in a separate directory is entirely irrelevant to this.

This is a misuderstanding of how copyright of translated dictionaries work.

As I said, there are two parts that can be copyrighted: the choice of which Klingon words are selected to the dictionary (the entry structure) and the choice of which words are used to translate the entries. The Finnish dictionary has a 1:1 correspondence to the boQwI' entry structure. However, that is not a problem, as the dictionary is significantly larger than the smaller TKD and KGT dictionaries and it doesn't matter that it includes all their entries. It is common that a larger dictionary may include all words of a smaller dictionary. One could disagree with me about this, but since the number of existing words is limited, there is no creativity involved in selecting the words and I'm willing to accept that risk.

The choice of translations itself, ie. the content of the entries, is also copyrightable. In this case the answer is easy: the words used in the Finnish translations are different than the English words (as they are not the same words). There is no 1:1 correspondence between English and Finnish lexemes. This is comparable to a situation where one copied the English dictionary and replaced some of the words with their synonyms. That would change their expression but not the meaning which cannot be copyrighted. That actually is how new dictionaries are constructed: one typically reads multiple dictionaries and other sources, chooses which words to include, and then takes some of the words from different sources and some from other sources. The resulting dictionary resembles the source dictionaries but not all entries are identical. Of course a lot of small entries with only one translation will end up to be identical, but if enough of them are different, it doesn't matter.

Of course, if one replaced each word in the Lord of the Rings with synonyms, that would just be a kind of encryption and the work would still be a copyright violation, since it would have the 1:1 correspondence in structure even if the words themselves are different. However, since I have explained that the structure of the dictionary is not a problem in this case, there is no problem with the translated entries either.

**Comment by De7vID** [comment]
> This is a misuderstanding of how copyright of translated dictionaries work.

The **boQwI'** database _is not a dictionary_. 

> This is comparable to a situation where one copied the English dictionary and replaced some of the words with their synonyms. That would change their expression but not the meaning which cannot be copyrighted. That actually is how new dictionaries are constructed: 

I would strongly advise you to check your understanding with a lawyer. Based on legal information I've been previously told about the status of this project, and the various translations within it, I don't think the situation is comparable to what you describe.

**Comment by fergusq** [comment]
> The boQwI' database is not a dictionary.

Yes it is. I'm not sure what you are talking about. Can you elaborate.

**Comment by fergusq** [comment]
After discussing with De'vID privately I'm convinced that the database does not violate copyright in the US. I was wrong to claim it was illegal and I apologize for the harm I caused with my statement.

I will make the edits to the rest of the scripts and make a new PR which contains the newest changes (since merging this with master would be hard). I'll also write instructions for using the new scripts.

**Comment by thiagocmatos** [comment]
@fergusq 
Do you have already a proposal on the workflow for adding new entries and highlithing updates on the english definition that needs to be considered on the translations?

**Comment by fergusq** [comment]
Closing this and creating a new PR with updated scripts and instructions

**Comment by De7vID** [comment]
I consulted with a lawyer about the legal status of **boQwI'** and the Apache license is valid, with myself as the licensor. The lawyer said that it's very clear under **American copyright law** (and also, incidentally, Canadian copyright law) that the copyright to the data belongs to me and that it would be CBS/Paramount who would have to license it from me, should they wish to do something with it. The lawyer also advised against changing the legal structure of the project as the way it is set up is perfectly clear. 

I am very hesitant to go against advice from an actual lawyer, so I'm disinclined to make the suggested changes and would strongly prefer to keep things as-is. If you believe there is a problem under Finnish law, or that the Finnish data should be covered by a different license for some reason, can you please confirm the legal consequences of the suggested changes with a lawyer who is familiar with both American and Finnish copyright law?

**Comment by fergusq** [comment]
The reason to have a different license isn't that I fear there is a problem. The reason is to protect my interests should CBS/Paramount want to use the data. Under the Apache License, they can use it free of charge and without contributing anything to the project, which I feel like taking advantage of the work I have done.

Normally I'd license a software project under the GPL license, which I feel is in sufficient balance between allowing free commercial and non-commercial use that forces the users to contribute back to the project, either by buying a different license or by giving back their changes to the code.

However, to my knowledge, there aren't similar licenses that work for databases. Due to this, I selected the CC BY-NC license that arguably is very restrictive, but it would prevent companies from taking advantage of the data.
